lib/pbench/server/filetree.py

```diff
@@ -339,40 +340,71 @@ def unpack(self, incoming: Path, results: Path):
 
         try:
             process = subprocess.run(
-                ["tar", "-xf", self.tarball_path],
+                [
+                    "tar",
+                    "-xf",
+                    "--no-same-owner",
+                    "--delay-directory-restore",
+                    "--force-local",
+                    self.tarball_path,
+                ],
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballUnpackError(self.tarball_path.name, str(exc)) from exc
+            raise TarballUnpackError(self.name, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballUnpackError(
-                    self.tarball_path.name,
+                    self.name,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        command = "find unpacked/ \( -type d -exec chmod ugo+rx '{}' \; \) -o \( -type f -exec chmod ugo+r '{}' \; \)"
+        command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
         try:
             process = subprocess.run(
-                command,
-                shell=True,
+                command.split(),
+                cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballModeChangeError(self.tarball_path.name, str(exc)) from exc
+            raise TarballModeChangeError(self.name, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballModeChangeError(
-                    self.tarball_path.name,
+                    self.name,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        self.unpacked = Path(str(unpacked)[: -len(".unpack")])
+        self.unpacked = incoming
```

Comment:
whole tarball directory is being moved from (incoming/f"{self.name}.unpack), so if I will be doing `incoming/self.name` it will be like `incoming/self.name/tarball_folder` . The tarball folders will be redundant.